File: milestone 2/python_modules/module04/ex0/ft_ancient_text.py
import sys

if __name__ == "__main__":
    argc: int = len(sys.argv)

    if argc > 1:
        try:
            filename: str = sys.argv[1]

            print("=== Cyber Archives Recovery ===")
            print(f"Accessing file '{filename}'")
            with open(filename, "r") as file:
                print("---\n\n")
                content = file.read()
                print(content)
                print("---")
                print(f"File '{filename}' closed.")
        except FileNotFoundError as e:
            print(f"Error opening file '{filename}': {e}")
    else:
        print("Usage: ft_ancient_text.py <file>")

File: milestone 2/python_modules/module04/ex0/ft_ancient_text.cpp
#include <iostream>
#include <cstring>
#include <string>
#include <fstream>
using namespace std;

class CustomError : public exception
{
    private:
        string full_message;
        int error_num;
    
    public:
    CustomError(int error_num, const string &message)
    : full_message("Error opening file: [Errno " + to_string(error_num) + "] " + message),
      error_num(error_num) {}

    const char* what() const noexcept override
    {
        return full_message.c_str();
    }
};

int main(int argc, char **argv)
{
    if (argc > 1)
    {
        string filename = argv[1];
        try
        {
            cout << "=== Cyber Archives Recovery ===\n";
            cout << "Accessing file '" << filename << "'" << endl;

            if (filename.find("/etc/") != string::npos)
                throw CustomError(13, " Permission denied: '" + filename + "'");
        }
        catch(const CustomError &e)
        {
            std::cerr << e.what() << '\n';
        }
        
    }
}